src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -178,6 +180,37 @@ internal T WithHttpManager(IHttpManager httpManager)
             return (T)this;
         }
 
+
+        /// <summary>
+        /// MSAL maintains a token cache in memory. By default, this cache object is part of each instance of <see cref="PublicClientApplication"/> or <see cref="ConfidentialClientApplication"/>
+        /// This method allows customization of the in-memory token cache of MSAL. 
+        /// 
+        /// MSAL's memory cache is different than token cache serialization. Cache serialization pulls the tokens from a cache (e.g. Redis, Cosmos or a file on disk), 
+        /// where they are stored in JSON format, into MSAL's memory cache. Memory cache operations do not involve JSON operations. 
+        /// 
+        /// External cache serialization remains the recommended way to handle desktop apps, web site and web apis, as it provides persistence.
+        /// 
+        /// Detailed guidance for each application type and platform:
+        /// https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnetcore
+        /// </summary>
+        /// <param name="useStaticCache">Share the static cache between all ClientApplication objects. Defaults to false.</param>
+        /// <remarks>The legacy ADAL library used a static cache by default.</remarks>
+#if !SUPPORTS_CUSTOM_CACHE || WINDOWS_APP
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#endif
+        public T WithMemoryTokenCacheOptions(bool useStaticCache)
+        {
+            // TODO: expose on options
```

Comment:
Is this important enough to add to options? I think it is, can action this tomorrow.